src/store/modules/ADempiere/field/search/invoice.js

```diff
@@ -0,0 +1,383 @@
+        }
+        const pageToken = generatePageToken({ pageNumber })
+
+        commit('setBusinessPartnerIsLoading', {
+          containerUuid,
+          isLoading: true
+        })
+
+        const isSalesTransactionContext = isSalesTransaction({
+          parentUuid: parentUuid,
+          containerUuid: containerUuid
+        })
+
+        const { queryFilters } = storedBusinessPartnerData
+        if (isSalesTransactionContext) {
+          queryFilters.is_vendor = undefined
+        } else {
+          queryFilters.is_customer = undefined
+        }
+        commit('setInvoiceFieldData', {
+          ...storedBusinessPartnerData,
+          containerUuid
+        })
+        requestListInvoicesInfo({
+          contextColumnNames,
+          // References
+          processParameterId,
+          browseFieldId,
+          referenceId,
+          columnName,
+          columnId,
+          fieldId,
+          // Query
+          searchValue,
+          ...queryFilters,
+          // Page Data
+          pageToken,
+          pageSize
+        })
+          .then(responseBusinessPartnerList => {
+            const recordsList = responseBusinessPartnerList.records.map((row, rowIndex) => {
+              return {
+                [COLUMN_NAME]: row.id,
+                ...row,
+                // datatables app attributes
+                ...ROW_ATTRIBUTES,
+                rowIndex
+              }
+            })
+
+            let currentRow = {}
+            // update current record
+            if (!isEmptyValue(recordsList)) {
+              // set first record
+              currentRow = recordsList.at(0)
+            }
+
+            commit('setInvoiceFieldData', {
+              ...storedBusinessPartnerData,
+              containerUuid,
+              currentRow,
+              recordsList,
+              nextPageToken: responseBusinessPartnerList.next_page_token,
+              pageNumber,
+              pageSize,
+              isSalesTransaction: isSalesTransactionContext,
+              isLoaded: true,
+              recordCount: Number(responseBusinessPartnerList.record_count)
+            })
+
+            resolve(recordsList)
+          })
+          .catch(error => {
+            console.warn(error)
+            showMessage({
+              type: 'info',
+              message: error.message
+            })
+          })
+          .finally(() => {
+            setTimeout(() => {
+              commit('setBusinessPartnerIsLoading', {
+                containerUuid,
+                isLoading: false
+              })
+            }, 500)
+          })
+      })
+    },
+    invoiceBusinessPartnerList({ commit }, searchValue) {
+      return new Promise(resolve => {
+        let list
+        requestListBusinessPartners({
+          search_value: searchValue,
+          page_size: 300
+        })
+          .then(response => {
+            const { records } = response
+            list = records.map((list) => {
```

Comment:
Remove this loop to improve peformance.

Reply:
https://github.com/solop-develop/frontend-core/pull/2220/commits/9531be9e2962182759a5db75cd377bcda4bb6ece